errors/field.go

```diff
@@ -0,0 +1,100 @@
+package errors
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+)
+
+// Field returns an error instance that wraps the original error with
+// additional information. It returns `nil` if provided error is `nil`.
+// Use this function to create an error instance describing a field/attribute
+// error.
+// This function might attach a stack trace information.
+func Field(fieldName string, err error, description string) error {
+	if isNilErr(err) {
+		return nil
+	}
+
+	// If this error does not carry the stacktrace information yet, attach
+	// one. This should be done only once per error at the lowest frame
+	// possible (most inner wrap).
+	if stackTrace(err) == nil {
+		err = errors.WithStack(err)
+	}
+
+	return &fieldError{
+		parent: err,
+		field:  fieldName,
+		desc:   description,
+	}
+}
+
+// AppendField is a shortcut function to club together error(s) with a given
+// field error.
+func AppendField(errorsOrNil error, fieldName string, fieldErrOrNil error) error {
+	return Append(errorsOrNil, Field(fieldName, fieldErrOrNil, ""))
+}
+
+type fieldError struct {
+	parent error
+	field  string
+	desc   string
+}
+
+func (err *fieldError) Error() string {
+	if err.desc == "" {
+		return fmt.Sprintf("field %q: %s", err.field, err.parent)
+	}
+	return fmt.Sprintf("field %q: %s: %s", err.field, err.desc, err.parent)
+}
+
+// Cause implements the causer interface.
+func (err *fieldError) Cause() error {
+	return err.parent
+}
+
+// Field implements fielder interface.
+func (err *fieldError) Field() string {
+	return err.field
+}
+
+// FieldErrors returns the list of all errors that are created for the given
+// field name.
+// An error must be implementing a fielder interface and return a matching
+// field name in order to pass the test and be included in the result set.
+func FieldErrors(err error, fieldName string) []error {
```

Comment:
pp: this methods name could be a bit more verbose like `ExtractFieldErrors` or `ParseFieldError` to not mix it with error creation and keep the good and short names for prod code.